ImportExport/ObjectsToPython/ObjectsToPython.FCMacro

```diff
@@ -0,0 +1,344 @@
+    addScriptFunc("from FreeCAD import Vector, Placement, Rotation")
+    addScriptFunc("import Sketcher");
+    addScriptFunc("import Part");
+    addScript("")
+    objectlist = []
+    skipObjects = [('App::Line', 'X_Axis'), ('App::Line', 'Y_Axis'), ('App::Line', 'Z_Axis'),
+                   ('App::Plane', 'XY_Plane'), ('App::Plane', 'XZ_Plane'), ('App::Plane', 'YZ_Plane'),
+                   ('App::Origin', 'Origin')] 
+    
+    selection = FreeCADGui.Selection.getSelection() 
+    if not selection:
+        selection = doc.Objects
+    else:
+        expandSelection(selection)
+        
+    for obj in doc.Objects:  
+        objectlist.append(doc.getObject(obj.Name)) 
+         
+    for obj in selection:    
+        if obj.TypeId == "Sketcher::SketchObject": 
+            addScriptFunc("")
+            addScriptFunc("def createSketch_%s(doc):"%varname(obj))
+            addObject(doc, obj, objectlist)
+            addScript("return " + varname(obj))
+        
+    addScriptFunc("")
+    addScriptFunc("")
+    addScriptFunc("def make_%s():"%doc.Name)
+    if not selection:
+        addScript("doc = App.newDocument('%s')"%doc.Name)
+    else:
+        addScript("doc = App.ActiveDocument")
+    
+    for obj in selection: 
+        if (obj.TypeId, obj.Label) not in skipObjects:
+            addScriptFunc("")
+            if obj.TypeId == "Sketcher::SketchObject":
+                addScript(varname(obj) + " = createSketch_%s(doc)"%varname(obj))
+            else:
+                addObject(doc, obj, objectlist)            
+                
+    for obj in selection: 
+        if obj.TypeId == 'PartDesign::Body':
+            addScriptFunc("")
+            addScript("%s.Group = %s_Group"%(varname(obj), varname(obj))) 
+            if obj.Tip:
+                addScript("%s.Tip = %s"%(varname(obj), varname(obj.Tip))) 
+            
+    addScript("") 
+    addScript("doc.recompute()") 
+    addScriptFunc("")
+    addScriptFunc("")
+    addScriptFunc("make_%s()"%doc.Name)
+        
+  
+def varname(obj):
+    return obj.Label.replace(" ", "_")
+      
+def addScript(line):
+    FreeCAD.Console.PrintMessage("\t" + line + "\n")
+    
+def addScriptFunc(line):
+    FreeCAD.Console.PrintMessage(line + "\n")
+    
+def expandSelection(selection):    
+    addthis = []     
+    for obj in selection:    
+        if hasattr(obj, 'Group'):
+            addthis += obj.Group
+        if hasattr(obj, 'Source'):
+            addthis.append(obj.Source)
+        if hasattr(obj, 'Links'):
+            addthis += obj.Links
+            
+    added = 0
+    for a in addthis:
+        if (not a in selection) and hasattr(obj, 'Name') and hasattr(obj, 'TypeId'):
+            selection.insert(0, a)
+            added = added + 1
+            
+    if added > 0:
+        expandSelection(selection)
+            
+
+def addObject(doc, obj, objectlist): 
+    objname = varname(obj)
+    addScript("%s = doc.addObject('%s', '%s')"%(objname, obj.TypeId, objname))
```

Comment:
I would prefer `addScript("{name} = doc.addObject('{type_}', '{name}')" .format(name=objname, type_=obj.TypeId)` (as long as f-strings shouldn't be used yet) but it's a question of personal preference.

Reply:
I am too lazy to change this, but I will do it this way in my next project.

Reply:
Is it OK with you if I do it myself?

Reply:
yes, sure